### PR TITLE
Adds Apollo tracing support into graphql-java

### PR DESCRIPTION
**Pull request by bbakerman**
See #576 for a full description

This is dependent on https://github.com/graphql-java/graphql-java/pull/575 being accepted since it requires the extra instrumentation info to work

### CODE REVIEW

**Comment by martijnwalraven** [comment]
Great to see this moving so fast! One point of feedback is that you may want to use `Instant.now()` instead of `ZonedDateTime.now()`, and `DateTimeFormatter.ISO_INSTANT` instead of `DateTimeFormatter.ISO_OFFSET_DATE_TIME` for `startTime` and `endTime`.

File: src/test/groovy/graphql/execution/instrumentation/TestingInstrumentation.groovy [hunk 1]
```diff
@@ -1,54 +1,63 @@
 package graphql.execution.instrumentation
 
 import graphql.ExecutionResult
-import graphql.execution.instrumentation.parameters.InstrumentationDataFetchParameters
-import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters
-import graphql.execution.instrumentation.parameters.InstrumentationFieldFetchParameters
-import graphql.execution.instrumentation.parameters.InstrumentationFieldParameters
-import graphql.execution.instrumentation.parameters.InstrumentationValidationParameters
+import graphql.execution.instrumentation.parameters.*
```
Comment (bbakerman) [inline src/test/groovy/graphql/execution/instrumentation/TestingInstrumentation.groovy]: Bah - some how I have changed my IDEA settings :(

**Comment by bbakerman** [comment]
@martijnwalraven - you might want to pass that tip onto the Sangria team since they also use DateTimeFormatter.ISO_OFFSET_DATE_TIME.  I looked at their code for inspiration

**Comment by bbakerman** [comment]
Ok moved to DateTimeFormatter.ISO_INSTANT

**Comment by martijnwalraven** [comment]
@bbakerman: Great! The Scala code has also been changed.

**Pull request merged**